smolyay/basis.py

```diff
@@ -1,34 +1,81 @@
 import abc
-import math
 
 import numpy
 import scipy.special
 
+
 class BasisFunction(abc.ABC):
     """Basis function for interpolating data.
 
-     A one-dimensional basis function is defined on the domain
-     :math:`[-1,1]`. The function defines the :attr:`points` at
-     which it should be sampled within this interval for interpolation.
-     The function also has an associated :meth:`__call__` method
-     for evaluating it at a point within its domain. Moreover,
-     the first derivative of the function can be evaluated via
-     :meth:`derivative`.
+    A one-dimensional basis function is defined on some given natural
+    domain. The function defines the :attr:`points` at which it should
+    be sampled within this interval for interpolation. The function also
+    has an associated :meth:`__call__` method for evaluating it at a
+    point within its domain. Moreover, the first derivative of the function
+    can be evaluated via :meth:`derivative`.
     """
 
-    def __init__(self):
-        self._points = []
-
     @property
     @abc.abstractmethod
-    def points(self):
-        """list: Sampling points for interpolation."""
+    def domain(self):
+        """list: Domain the sample points come from."""
```

Comment:
In refactoring our code, I would vote that we prefer to use NumPy arrays as much as possible, with exceptions for cases where it is not possible (e.g., lists of lists of unequal size)
```suggestion
        """numpy.ndarray: Domain the sample points come from."""
```